bitmark-references/bitmark-node-software/bitmark-blockain.md

```diff
@@ -0,0 +1,125 @@
+
+Bitmark uses [argon2](https://en.wikipedia.org/wiki/Argon2) to hash the block header; remaining data is covered by having the Merkle Tree Root hash as part of the header. If transactions are different or in a different order the the Merkle Root has will not match and the block will be rejected.
+
+#### Majority Votes
+
+When deciding next block, every node considers decisions from other nodes, the most block selected by other nodes will be chosen. Every 30 seconds, a node asks its connected nodes for their block hash and block height. This information will be used to determine majority votes. For a specific chain, same block hegith exists same block hash, so a node can know itself status by this information.
+
+Some conditionas could happen when deciding block:
+
+1. majority votes exists
+
+  Node    Block Height    Block Hash
+ ------  --------------  ------------
+  A       1000            abcdefg
+  B       1000            abcdefg
+  C       1000            gfedcba
+  D       1000            abcdefg
+  E       1000            abcdefg
+  F       1000            abcdefg
+
+In this example, node C resides on different chain than other (because
+its block hash is different from others at same block height).
+
+For node A, when it asks other nodes situation with following response
+(assumes all nodes are connected):
+
+  Count    Block Height    Block Hash
+ -------  --------------  ------------
+  4        1000            abcdefg
+  1        1000            gfedcba
+
+So node A will keep choose chain that has hash abcdefg on block
+height 1000.
+
+For node C, when it asks other nodes situation with following response
+(assumes all nodes are connectd):
+
+  Count   Block Height   Block Hash
+ ------- -------------- ------------
+  5       1000           abcdefg
+
+So node C will choose chain that has hash abcdefg on block height
+1000, which means it selft will sync to other nodes.
+
+1. several groups with same votes
+
+  Node   Block Height   Block Hash
+ ------ -------------- ------------
+  A      1000           abcdefg
+  B      1000           abcdefg
+  C      1000           abcdefg
+  D      1000           gfedcba
+  E      1000           gfedcba
+  F      1000           gfedcba
+
+In thie example, two groups of nodes with equal votes. If there's a
+new node G connect into network, the response from G as follows:
+
+  Count   Block Height   Block Hash
+ ------- -------------- ------------
+  3       1000           abcdefg
+  3       1000           gfedcba
+
```

Comment:
I think the more relevant question is what happens if NO new node is added. Does the tie remain, or does it naturally move toward the lower hash criterion?

Reply:
If no new node connected and current network stays in a balance, every node uses digest to compare, as example listed.

Reply:
I suggest just giving that example, rather than making things more complex by suddenly adding a new node.